.travis.yml

```diff
@@ -198,7 +199,14 @@ install:
           esac
         - |
           if [ -n "$TARGET" ] ; then
-           if [ ! -f "dpdk/${TARGET}/lib/libdpdk.a" ] ; then
+           # force DPDK rebuild if it does not match shared/non-shared requirement
+           if [ -n "$DPDK_SHARED" -a ! -f "dpdk/${TARGET}/lib/libdpdk.so" ] ; then
+            rm -rf dpdk/"$TARGET"
+           fi
+           if [ -z "$DPDK_SHARED" -a ! -f "dpdk/${TARGET}/lib/libdpdk.a" ] ; then
+            rm -rf dpdk/"$TARGET"
+           fi
```

Comment:
maybe simple  
`if [  -a ! -f "dpdk/${TARGET}/lib/libdpdk.a" ] ; then`
instead of 2 checks?

Reply:
In general that is not right solution. I think we need to cache 2 dpdk builds. Shared and non shared and force dpdk recompilation only on version change.

Reply:
Hmm. I can try doing that.